test/HttpTest.cpp

```diff
@@ -37,44 +38,69 @@ TEST(HttpServer, HttpTest) {
     co_return co_await send(
         createOkResponse(response, request, ad_utility::MediaType::textPlain));
   };
-  // Set up a HTTP server and run it.
-  // TODO: Try out random ports until we find a free one.
-  short unsigned int port = 37654;
-  const std::string& ipAdress = "0.0.0.0";
-  int numServerThreads = 1;
-  auto httpServer = HttpServer{port, ipAdress, numServerThreads,
-                               std::move(mirroringHttpSessionHandler)};
-  std::jthread httpServerThread([&]() { httpServer.run(); });
+  // Set up a HTTP server and run it. Try out 10 different ports, if connection
+  // to all of them fail, the test fails.
+  //
+  // TODO: Is there a more robust way to do this? Should we try out more ports?
+  auto httpServer = [&mirroringHttpSessionHandler]() {
+    std::vector<short unsigned int> ports(10);
+    std::generate(ports.begin(), ports.end(),
+                  []() { return 1024 + std::rand() % (65535 - 1024); });
+    LOG(INFO)
+        << "Trying to start a test HTTP server on one of the following ports: "
+        << absl::StrJoin(ports, ", ") << std::endl;
+    const std::string& ipAddress = "0.0.0.0";
+    int numServerThreads = 1;
+    for (const short unsigned int port : ports) {
+      try {
+        return HttpServer{port, ipAddress, numServerThreads,
+                          std::move(mirroringHttpSessionHandler)};
```

Comment:
I understand + thanks for pointing this out. It sounds like a frequent use case. But is this the most elegant way to solve this? It's a bit paradoxical because lambdas were introduced so that one does not have to define classes with a single operator function and then instantiate objects of that class. And now we are turning a lambda (which implicitly is the operator function of a class) into a class and explicitly instantiate an object from that class.

Reply:
Well, the alternative is to explicitly define the lambda-like class with a call operator. A function would not work, as we need something overloaded/templated, and thus we wouldn't be able to get a unique address to pass to the `HttpServer`.
Also note, that since you're not capturing, you could also define this thing (lambda or class) outside the test function.

But what you could do:
Your lambda shoudl be copyable, so you could just leave everything as is and drop the `std::move()`.
Thanks for being persistent.